deepchem/models/torch_models/kfac_optimizer.py

```diff
@@ -1,486 +1,486 @@
-                    m.bias.grad.data.copy_(v[1])
-                    m.bias.grad.data.mul_(nu)
-                else:
-                    raise TypeError("bias.grad.data should be a Tensor")
-
-    def _step(self, closure: Optional[Callable] = None):
-        """
-        Called in every step of the optimizer, updating the model parameters from the gradient by the KFAC equation.
-        Also, performs weight decay and adds momentum if any.
-
-        Parameters:
-        -----------
-        closure: Callable, optional(default: None)
-            an optional customizable function to be passed which can be used to clear the gradients and other compute loss for every step.
-        """
-        for group in self.param_groups:
-            weight_decay = group['weight_decay']
-            momentum = group['momentum']
-            lr = group['lr']
-
-            for p in group['params']:
-                if p.grad is None:
-                    continue
-                d_p = p.grad.data
-                if weight_decay != 0 and self.steps >= 20 * self.TCov:
-                    d_p.add_(weight_decay, p.data)
-                if momentum != 0:
-                    param_state = self.state[p]
-                    if 'momentum_buffer' not in param_state:
-                        buf = param_state['momentum_buffer'] = torch.zeros_like(
-                            p.data)
-                        buf.mul_(momentum).add_(d_p)
-                    else:
-                        buf = param_state['momentum_buffer']
-                        buf.mul_(momentum).add_(d_p)
-                    d_p = buf
-
-                torch.add(p.data, -lr, d_p, out=p.data)
-
-    def step(self, closure: Optional[Callable] = None):
-        """
-        This is the function that gets called in each step of the optimizer to update the weights and biases of the model.
-
-        Parameters:
-        -----------
-        closure: Callable, optional(default: None)
-            an optional customizable function to be passed which can be used to clear the gradients and other compute loss for every step.
-        """
-        group = self.param_groups[0]
-        lr = group['lr']
-        damping = group['damping']
-        updates = {}
-        for m in self.modules:
-            if self.steps % self.TInv == 0:
-                self._update_inv(m)
-            p_grad_mat = self._get_matrix_form_grad(m)
-            v = self._get_natural_grad(m, p_grad_mat, damping)
-            updates[m] = v
-        self._kl_clip_and_update_grad(updates, lr)
-
-        self._step(closure)
-        self.steps += 1
+"""
+Implementation of KFAC, a second order optimizer, in PyTorch.
+"""
+import math
+from typing import Optional, Callable, Dict, Tuple, List, Union
+try:
+    import torch
+    import torch.optim as optim
+    has_pytorch = True
+
+except ModuleNotFoundError:
+    has_pytorch = False
+
+
+class KFACOptimizer(optim.Optimizer):

```

Comment:
I am not sure here why by changing a single typo, git is showing diff in many other lines, though the visible difference between other lines seems to be none.

Reply:
Yes, I am not able to find anything online for why it's happening.